### PR TITLE
Fix bug with nullable child in CheckboxGroup

### LINKED ISSUE #42: Cannot read property '$$typeof' of null
If I'll do something like this(one of the child is `null`):
```javascript
<CheckboxGroup checkboxDepth={3} name="fruit">
  <label><Checkbox value="kiwi"/> Kiwi</label>
  {false && ( // or another logical condition which returns `false`
    <label><Checkbox value="watermelon"/> Watermelon</label>
  )}
</CheckboxGroup>
```

I'll have an error: `TypeError: Cannot read property '$$typeof' of null`

### PR DESCRIPTION
**Pull request by bipycxp**
- Fixed bug with nullable child in CheckboxGroup
- Added tests
- Updated version to 4.0.1

resolve #42

### CODE REVIEW

File: react-checkbox-group.jsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class CheckboxGroup extends Component {
     };
 
     return React.Children.map(children, child => {
-      if (!child.$$typeof) {
+      if (!child || !child.$$typeof) {
```
Comment (ziad-saab) [inline react-checkbox-group.jsx]: Thanks for PR! Can you please change this line to:

```js
if (!(child && child.$$typeof))
```

File: test/react-checkbox-group_spec.js [hunk 2]
```diff
@@ -25,11 +25,13 @@ describe('ReactCheckboxGroup', function() {
     }).to.throw();
   });
 
-  it('Renders correctly when `Checkbox` is not a direct child of `CheckboxGroup` and `checkboxDepth` is passed', function() {
+  it('Renders correctly when `Checkbox` components on a different nesting levels and one of the child is null', function() {
```
Comment (ziad-saab) [inline test/react-checkbox-group_spec.js]: Thanks also for updating the tests! Can you please change this to be a separate test case for the `null`?

**Comment by bipycxp** [comment]
@ziad-saab Done ;)

**Comment by ziad-saab** [comment]
Thank you! Published as 4.0.1
